pgjdbc/src/test/java/org/postgresql/test/jdbc2/IntervalTest.java

```diff
@@ -375,6 +378,23 @@ public void testMicroSecondsAreRoundedToNearest() throws SQLException {
     assertEquals(1, pgi.getMicroSeconds());
   }
 
+  @Test
+  public void testConverter() throws SQLException {
+    PGInterval fromString = new PGInterval("+2004 years -4 mons +20 days -15:57:12.100001");
+    Period period = fromString.getPeriodPart();
+    Period expectedPeriod = Period.of(2004, -4, 20);
+    assertEquals(expectedPeriod, period);
+
+    Duration duration = fromString.getDurationPart();
+    Duration expectedDuration = Duration.ofHours(-15).plusMinutes(-57).plusSeconds(-12).plus(-100001, ChronoUnit.MICROS);
+    assertEquals(expectedDuration, duration);
+
+    assertEquals(fromString, PGInterval.from(expectedPeriod, expectedDuration));
+
+    PGInterval pgi = new PGInterval("0.0000007 seconds");
+    assertEquals(Duration.ofNanos(1000), pgi.getDurationPart());
```

Comment:
This seems to be a different test. Would you please put it in its own method?

Reply:
Changed.